test/drenv/envfile.py

```diff
@@ -13,58 +14,69 @@
 CONTAINER = "$container"
 SHARED_NETWORK = "$network"
 
+X86_64 = "x86_64"
+ARM64 = "arm64"
+
 _PLATFORM_DEFAULTS = {
     "__default__": {
         PROVIDER: {
-            "x86_64": "",
-            "arm64": "",
+            X86_64: "",
+            ARM64: "",
         },
         VM: {
-            "x86_64": "",
-            "arm64": "",
+            X86_64: "",
+            ARM64: "",
         },
         CONTAINER: "",
         SHARED_NETWORK: {
-            "x86_64": "",
-            "arm64": "",
+            X86_64: "",
+            ARM64: "",
         },
     },
     "linux": {
         PROVIDER: {
-            "x86_64": "minikube",
-            "arm64": "",
+            X86_64: "minikube",
+            ARM64: "",
         },
         VM: {
-            "x86_64": "kvm2",
-            "arm64": "",
+            X86_64: "kvm2",
+            ARM64: "",
         },
         CONTAINER: "docker",
         SHARED_NETWORK: {
-            "x86_64": "default",
-            "arm64": "",
+            X86_64: "default",
+            ARM64: "",
         },
     },
     "darwin": {
         PROVIDER: {
-            "x86_64": "lima",
-            "arm64": "lima",
+            X86_64: "lima",
+            ARM64: "lima",
         },
         VM: {
-            "x86_64": "",
-            "arm64": "",
+            X86_64: "",
+            ARM64: "",
         },
         CONTAINER: "podman",
         SHARED_NETWORK: {
-            "x86_64": "",
-            "arm64": "",
+            X86_64: "",
+            ARM64: "",
         },
     },
 }
 
 
+@cache
+def get_os_info():
+    # Returns cached system info tuple(str, str) after first call
+    operating_system = platform.system().lower()
+    machine = os.uname().machine
+    return operating_system, machine
+
+
 def platform_defaults():
     # By default, use provider defaults.
-    operating_system = platform.system().lower()
+    operating_system, _ = get_os_info()
     logging.debug("[envfile] Detected os: '%s'", operating_system)
```

Comment:
This change does not fix the problem, which is logging os and machine values multiple times. To avoid the duplicate logs, we need to detect the os and machine *once* and log the value when we detect them in the cached helper. The code code using the values should not log them again.